staticaddr/deposit/manager.go

```diff
@@ -420,3 +422,88 @@ func (m *Manager) finalizeDeposit(outpoint wire.OutPoint) {
 	delete(m.deposits, outpoint)
 	m.Unlock()
 }
+
+// GetActiveDepositsInState returns all active deposits.
+func (m *Manager) GetActiveDepositsInState(stateFilter fsm.StateType) (
+	[]*Deposit, error) {
+
+	m.Lock()
+	defer m.Unlock()
+
+	var deposits []*Deposit
+	for _, fsm := range m.activeDeposits {
+		if fsm.deposit.GetState() != stateFilter {
+			continue
+		}
+		deposits = append(deposits, fsm.deposit)
+	}
+
+	sort.Slice(deposits, func(i, j int) bool {
+		return deposits[i].ConfirmationHeight <
+			deposits[j].ConfirmationHeight
+	})
+
+	return deposits, nil
+}
+
+// GetAllDeposits returns all active deposits.
+func (m *Manager) GetAllDeposits() ([]*Deposit, error) {
+	return m.cfg.Store.AllDeposits(m.runCtx)
+}
+
+// AllOutpointsActiveDeposits checks if all deposits referenced by the outpoints
+// are active and in the specified state.
+func (m *Manager) AllOutpointsActiveDeposits(outpoints []wire.OutPoint,
+	stateFilter fsm.StateType) ([]*Deposit, bool) {
+
+	m.Lock()
+	defer m.Unlock()
+
+	deposits := make([]*Deposit, 0, len(outpoints))
+	for _, o := range outpoints {
+		if _, ok := m.activeDeposits[o]; !ok {
+			return nil, false
+		}
+
+		deposit := m.deposits[o]
+		if deposit.GetState() != stateFilter {
+			return nil, false
+		}
+
+		deposits = append(deposits, m.deposits[o])
+	}
+
+	return deposits, true
+}
+
+// TransitionDeposits allows a caller to transition a set of deposits to a new
+// state.
+func (m *Manager) TransitionDeposits(deposits []*Deposit, event fsm.EventType,
+	expectedFinalState fsm.StateType) error {
+
+	for _, d := range deposits {
+		m.Lock()
+		sm, ok := m.activeDeposits[d.OutPoint]
+		m.Unlock()
+		if !ok {
+			return fmt.Errorf("deposit not found")
+		}
+
+		err := sm.SendEvent(event, nil)
+		if err != nil {
+			return err
+		}
+		err = sm.DefaultObserver.WaitForState(
```

Comment:
Do we expect that the one minute will be enough? Perhaps could be a timeout param?